augur/tasks/util/collection_util.py

```diff
@@ -425,3 +433,155 @@ def start_block_of_repos(logger,session,condition,limit,phases,repos_type,hook="
     routine.start_data_collection()
 
     return len(repo_git_identifiers)
+
+"""
+    Generalized function for starting a phase of tasks for a given collection hook with options to add restrictive conditions
+"""
+def start_repos_by_user(session, max_repo,phase_list, days_until_collect_again = 1, hook="core",new_status=CollectionState.PENDING.value,additional_conditions=None):
+
+    #getattr(CollectionStatus,f"{hook}_status" ) represents the status of the given hook
+    #Get the count of repos that are currently running this collection hook
+    desired_status = f"{hook}_status"
+    active_repo_count = len(session.query(CollectionStatus).filter(getattr(CollectionStatus,desired_status ) == CollectionState.COLLECTING.value).all())
+
```

Comment:
I think changing `desired_status` to `status_column` would be more clear.

Reply:
That makes sense